core/src/raw/adapters/kv/api.rs

```diff
@@ -18,17 +18,28 @@
 use std::fmt::Debug;
 use std::future::ready;
 
+use futures::stream::Empty;
 use futures::Future;
+use futures::Stream;
 
 use crate::raw::*;
 use crate::Capability;
 use crate::Scheme;
 use crate::*;
 
+/// A noop placeholder for Adapter::ScanIter
+pub type EmptyScanIter = Empty<Result<String>>;
+
 /// KvAdapter is the adapter to underlying kv services.
 ///
 /// By implement this trait, any kv service can work as an OpenDAL Service.
 pub trait Adapter: Send + Sync + Debug + Unpin + 'static {
+    /// async iterator type for Adapter::scan()
+    ///
+    /// TODO: consider to replace it with std::async_iter::AsyncIterator after stablized
+    /// TODO: use default associate type `= EmptyScanIter` after stablized
+    type ScanIter: Stream<Item = Result<String>> + Send + Unpin;
```

Comment:
Hi, OpenDAL tends not to depend directly on those traits. Instead, we can define our own `kv::Scan`, `kv::BlockingScan` like we do for [`oio::List`](https://github.com/apache/opendal/blob/a12457912d6aa2d22039fb016ecd42f93166d83e/core/src/raw/oio/list/api.rs#L29-L35)

Reply:
Done.